dm/worker/config.go

```diff
@@ -266,26 +315,40 @@ func (c *Config) adjustFlavor() error {
 			return terror.ErrNotSupportedFlavor.Generate(c.Flavor)
 		}
 	}
-	// decrypt password
-	clone, err := c.DecryptPassword()
-	if err != nil {
-		return err
+
+	c.Flavor, err = utils.GetFlavor(ctx, db)
+	if ctx.Err() != nil {
+		err = terror.Annotatef(err, "time cost to get flavor info exceeds %s", dbGetTimeout)
+	}
+	return terror.WithScope(err, terror.ScopeUpstream)
+}
+
+func (c *Config) adjustServerID(ctx context.Context, db *sql.DB) error {
+	if c.ServerID != 0 {
+		return nil
+	}
+
+	serverIDs, err := getAllServerIDFunc(ctx, db)
+	if ctx.Err() != nil {
+		err = terror.Annotatef(err, "time cost to get server-id info exceeds %s", dbGetTimeout)
 	}
-	from := clone.From
-	from.RawDBCfg = config.DefaultRawDBConfig().SetReadTimeout(flavorReadTimeout)
-	fromDB, err := applyNewBaseDB(from)
 	if err != nil {
 		return terror.WithScope(err, terror.ScopeUpstream)
 	}
-	defer fromDB.Close()
 
-	ctx, cancel := context.WithTimeout(context.Background(), flavorGetTimeout)
-	defer cancel()
-	c.Flavor, err = utils.GetFlavor(ctx, fromDB.DB)
-	if ctx.Err() != nil {
-		err = terror.Annotatef(err, "time cost to get flavor info exceeds %s", flavorGetTimeout)
+	rand.Seed(time.Now().UnixNano())
+	for i := 0; i < 5; i++ {
+		randomValue := uint32(rand.Intn(100000))
+		randomServerID := maxServerID/10 + randomValue
```

Comment:
It sames that the only use of `maxServerID` is `maxServerID/10`. How about set `maxServerID` directly to `math.MaxUint32/10`?

Reply:
update 354a2fc

Reply:
What about store `math.MaxUint32/10` into a variable so that we don't need to compute it every time?

Reply:
@WangXiangUSTC

Reply:
done 9b89300